components/server/ee/src/prebuilds/prebuild-manager.ts

```diff
@@ -58,6 +58,22 @@ export class PrebuildManager {
     @inject(Config) protected readonly config: Config;
     @inject(ProjectsService) protected readonly projectService: ProjectsService;
 
+    async abortPrebuildsForBranch(ctx: TraceContext, project: Project, user: User, branch: string): Promise<void> {
+        const span = TraceContext.startSpan("abortPrebuildsForBranch", ctx);
+        const prebuilds = await this.workspaceDB
+            .trace({ span })
+            .findActivePrebuiltWorkspacesByBranch(project.id, branch);
+        const results: Promise<any>[] = [];
+        for (const prebuild of prebuilds) {
+            for (const instance of prebuild.instances) {
+                results.push(this.workspaceStarter.stopWorkspaceInstance({ span }, instance.id, instance.region));
+            }
+            prebuild.prebuild.state = "aborted";
```

Comment:
Please also specify the cancellation reason by setting a helpful `prebuild.error` value, similar to the [rate-limit error](https://github.com/gitpod-io/gitpod/blob/7d005012a82778e5b916bf494a2c64bd461fec7d/components/server/ee/src/prebuilds/prebuild-manager.ts#L188-L191):

```suggestion
            prebuild.prebuild.state = "aborted";
            prebuild.error = "A newer commit was pushed to the same branch.";
```